plugin/core/settings.py

```diff
@@ -1,176 +1,68 @@
 from .collections import DottedDict
 from .logging import debug
-from .types import Settings, ClientConfig, LanguageConfig, syntax2scope
-from .typing import Any, List, Optional, Dict, Callable, Union
-from copy import deepcopy
+from .types import ClientConfig
```

Comment:
Is there are reason for `ClientConfig` being in `types.py`, separate from `ClientConfigs`?
It feels a bit wrong to have `ClientConfig` in `types.py`, I feel because types.py should in theory by for... types :)